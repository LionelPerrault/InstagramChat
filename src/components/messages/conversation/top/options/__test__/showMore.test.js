import React from 'react'
import { create } from 'react-test-renderer'
import ConversationShowMore from '../showMore'
import { shallow } from 'enzyme'

describe('ConversationShowMore Component', () => {
  const mockFn = jest.fn()
  const props = {
    toggleOptions: mockFn,
  }

  it('should match snapshot', () => {
    const tree = create(<ConversationShowMore {...props} />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('should show <AboutConversation/> when clicked on more option', () => {
    const wrapper = shallow(<ConversationShowMore {...props} />)
    wrapper.find('.m_m_info').simulate('click', {
      preventDefault() {},
    })
    expect(wrapper.find('Connect(AboutConversation)').exists()).toBe(true)
  })
})
