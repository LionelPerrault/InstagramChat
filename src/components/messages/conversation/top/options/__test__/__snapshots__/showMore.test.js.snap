// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConversationShowMore Component should match snapshot 1`] = `
<li>
  <a
    className="m_m_info"
    href="#"
    onClick={[Function]}
  >
    More
  </a>
</li>
`;
